py/server/deephaven/table.py

```diff
@@ -1299,11 +1310,56 @@ def agg_by(self, aggs: Union[Aggregation, Sequence[Aggregation]], by: Union[str,
         try:
             aggs = to_sequence(aggs)
             by = to_sequence(by)
+            if not by and initial_groups:
+                raise ValueError("missing group-by column names when initial_groups is provided.")
             j_agg_list = j_array_list([agg.j_aggregation for agg in aggs])
-            return Table(j_table=self.j_table.aggBy(j_agg_list, *by))
+            if not by:
+                return Table(j_table=self.j_table.aggBy(j_agg_list, preserve_empty))
+            else:
+                j_column_name_list = j_array_list([_JColumnName.of(col) for col in by])
+                initial_groups = unwrap(initial_groups)
+                return Table(
+                    j_table=self.j_table.aggBy(j_agg_list, preserve_empty, initial_groups, j_column_name_list))
         except Exception as e:
             raise DHError(e, "table agg_by operation failed.") from e
 
+    def partitioned_agg_by(self, aggs: Union[Aggregation, Sequence[Aggregation]],
+                           by: Union[str, Sequence[str]] = None, preserve_empty: bool = False,
+                           initial_groups: Table = None) -> PartitionedTable:
+        """The partitioned_agg_by method is a convenience method that performs an agg_by operation on this table and
+        wraps the result in a PartitionedTable. If the argument 'aggs' does not include a partition aggregation
+        created by calling :py:func:`agg.partition`, one will be added automatically with the default constituent column
+        name __CONSTITUENT__.
```

Comment:
I'm unclear here.

Does `t.partitioned_agg_by(aggs=[agg.sum("X")],by=["Y"])` create a partitioned table containing 3 columns ("X", "Y", and "`__CONSTITUENT__`", while `t.partitioned_agg_by(aggs=[agg.sum("X"), agg.partition("Z")],by=["Y"])` creates a table containing 3 columns ("X", "Y", "Z")?

What is the use case for this?

Reply:
I don't think you're too unclear, since you got it exactly right. The use case is to avoid having to have users construct their own `PartitionedTable` when they do other aggregations in addition to the partition.